code/modules/multiz/movement.dm

```diff
@@ -36,10 +36,19 @@
 		return eyeobj.zMove(direction)
 
 	// Check if we can actually travel a Z-level.
-	if (!can_ztravel())
+	if (!can_ztravel(direction))
 		to_chat(src, "<span class='warning'>You lack means of travel in that direction.</span>")
 		return FALSE
 
+	if (isliving(src))
```

Comment:
Remove the `/mob/living/can_ztravel(var/direction)` override and do this:

```
/mob/living/zMove(direction)
	if (is_ventcrawling)
		var/obj/machinery/atmospherics/pipe/zpipe/P = loc
		if (istype(P) && P.can_z_crawl(src, direction))
			return P.handle_z_crawl(src, direction)

	return ..()
```

The benefit is surprising. It's actually half the size in terms of the logic.